whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestBasicAuthTestIntegration.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    @Test
+    public void lookup_correct_basic_auth_password_and_unfiltered() {
+        databaseHelper.addObject(TEST_IRT);
+
+        final WhoisResources whoisResources = SecureRestTest.target(getSecurePort(), "whois/test/irt/irt-test?unfiltered")
+                .request()
+                .header("Authorization", getBasicAuthenticationHeader("test", "test"))
+                .get(WhoisResources.class);
+
+        assertThat(whoisResources.getErrorMessages(), is(empty()));
+        assertThat(whoisResources.getWhoisObjects(), hasSize(1));
+        final WhoisObject whoisObject = whoisResources.getWhoisObjects().get(0);
+        assertThat(whoisObject.getAttributes(), contains(
+                new Attribute("irt", "irt-test"),
+                new Attribute("address", "RIPE NCC"),
+                new Attribute("e-mail", "noreply@ripe.net"),
+                new Attribute("admin-c", "TP1-TEST", null, "person", Link.create("http://rest-test.db.ripe.net/test/person/TP1-TEST"), null),
+                new Attribute("tech-c", "TP1-TEST", null, "person", Link.create("http://rest-test.db.ripe.net/test/person/TP1-TEST"), null),
+                new Attribute("auth", "MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/", "test", null, null, null),
+                new Attribute("mnt-by", "OWNER-MNT", null, "mntner", Link.create("http://rest-test.db.ripe.net/test/mntner/OWNER-MNT"), null),
+                new Attribute("source", "TEST")));
+    }
+
+    @Test
+    public void create_succeeds_with_basic_auth_no_password_query() {
+        final WhoisResources whoisResources = SecureRestTest.target(getSecurePort(), "whois/test/person")
+                .request()
+                .header("Authorization", getBasicAuthenticationHeader("test", "test"))
+                .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), WhoisResources.class);
+
+        assertThat(whoisResources.getLink().getHref(), is(String.format("https://localhost:%s/test/person",getSecurePort())));
+        assertThat(whoisResources.getErrorMessages(), is(empty()));
+        final WhoisObject object = whoisResources.getWhoisObjects().get(0);
+
+        assertPersonObject(whoisResources, object);
+    }
+
+    @Test
+    public void create_failed_with_basic_auth_incorrect_www_authenticate_cookie() {
+        final Response response = SecureRestTest.target(getSecurePort(), "whois/test/person")
+                .request()
+                .header("Authorization", getBasicAuthenticationHeader("test", "incorrect"))
+                .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), Response.class);
+
+        assertThat(response.getHeaderString(HttpHeaders.WWW_AUTHENTICATE), is(BASIC_CHARSET_ISO_8859_1_LATIN_1));
+    }
+
+    @Test
+    public void create_failed_with_no_basic_auth_incorrect_password_no_www_authenticate_cookie() {
+        final Response response = SecureRestTest.target(getSecurePort(), "whois/test/person?password=incorrect")
```

Comment:
This should thrown an  NotAuthorizedException, right?

it is not better doing 

 NotAuthorizedException badAuth = assertThrows(NotAuthorizedException.class, () -> {
}

assertThat(badAuth.WWW_AUTH, is null)?

Reply:
I will add an assert for response code.

Reply:
I really prefer to have a common way of testing errors.
Sometimes we are checking the error directly in the call, another times asserting the code, some times using assertThrows...
I would say to use one way and not multiple of them.
If we say that we will use assert for the response always I will go with that as well. 
My only concern is to do this in different ways through the code

Reply:
But the purpose is to check WWW_authenticate header  is not there not to check exceptions. But may be you are correct, I do not mind assertThrows as well